### PR TITLE
build(wasm-builder): Propagate toolchain to wasm-builder

### LINKED ISSUE #2623: gear-wasm-builder: Add support for custom nightly toolchain when building contracts
### Problem to Solve

Need to have a possibility to choose a custom nightly toolchain when building smart contracts using `gear-wasm-builder`.

- Using `rust-toolchain.toml` which allows setting toolchain version:
    ```toml
    [toolchain]
    channel = "nightly-2023-03-14"
    targets = [
      "wasm32-unknown-unknown",
    ]
    ```
- Using `+nightly-<date>` in command line with `cargo build`:
    ```shell
    cargo +nightly-2023-03-14 build --release
    ```

### Possible Solution

Use env variables or parse `rust-toolchain.toml` inside `gear-wasm-builder`.

### Notes

_No response_

### PR DESCRIPTION
**Pull request by DennisInSky**
Resolves #2623.

Original toolchain passed to cargo using one of the [options](https://rust-lang.github.io/rustup/overrides.html#overrides) transformed and propagated down to the build script (wasm-builder).
The transformation rule replaces any channel by the `nightly` one, e.g.
`stable[-date]` -> `nightly[-date]`
`beta[-date]` -> `nightly[-date]`
`major.minor[-date]` -> `nightly[-date]`

@gear-tech/dev

### CODE REVIEW

File: utils/wasm-builder/src/cargo_toolchain.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
+// This file is part of Gear.
+
+// Copyright (C) 2022 Gear Technologies Inc.
```
Comment (shamilsan) [inline utils/wasm-builder/src/cargo_toolchain.rs:3]:
```suggestion
// Copyright (C) 2023 Gear Technologies Inc.
```

**Review by shamilsan — APPROVED** [review]
Awesome ❤️ 

Have you tried it with the [toolchain file](https://rust-lang.github.io/rustup/overrides.html#the-toolchain-file)?

**Comment by DennisInSky** [comment]
> Awesome ❤️
> 
> Have you tried it with the [toolchain file](https://rust-lang.github.io/rustup/overrides.html#the-toolchain-file)?

I have tried it and everything was propagated properly. Essentially, either options listed here https://rust-lang.github.io/rustup/overrides.html#overrides will work w/o any issues

File: utils/wasm-builder/src/cargo_toolchain.rs (hunk 1 above)
Comment (NikVolf) [reply utils/wasm-builder/src/cargo_toolchain.rs:3]: should be 2022-2023

**Pull request merged**